src/benchmarklib/synthetic_table_generator.cpp

```diff
@@ -62,7 +62,8 @@ std::shared_ptr<Table> SyntheticTableGenerator::generate_table(
     const std::vector<ColumnDataDistribution>& column_data_distributions,
     const std::vector<DataType>& column_data_types, const size_t num_rows, const ChunkOffset chunk_size,
     const std::optional<ChunkEncodingSpec>& segment_encoding_specs,
-    const std::optional<std::vector<std::string>>& column_names, const UseMvcc use_mvcc) {
+    const std::optional<std::vector<std::string>>& column_names, const UseMvcc use_mvcc,
+    const std::optional<float> null_ratio) {
```

Comment:
Just short (I just sent a rather concrete email): if you do it that way, please use rather a std::vector with a value for each column. You don't need to use make NULL optional, you could default to 0.0f.

Reply:
@Bouncner Thanks for your very valuable feedback! We created a separate issue (#8) for implementing the proposed changes to the method signature of SyntheticTableGenerator. 🙂